docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -698,6 +802,118 @@ With this option, the `Summary` property of the `WeatherForecastWithDefault` obj
 
 Null values in the JSON are ignored only if they are valid. Null values for non-nullable value types cause exceptions.
 
+## Preserve references and handle circular references
+
+::: zone pivot="dotnet-5-0"
+
+To preserve references and handle circular references, set <xref:System.Text.Json.JsonSerializerOptions.ReferenceHandler%2A> to <xref:System.Text.Json.Serialization.ReferenceHandler.Preserve%2A>. This setting causes the following behavior:
+
+* On serialize:
+
+  When writing complex types, the serializer also writes metadata properties (`$id`, `$values`, and `$ref`).
+
+* On deserialize:
+
+  Metadata is expected (although not mandatory), and the deserializer tries to understand it.
+
+The following code illustrates use of the `Preserve` setting.
+
+:::code language="csharp" source="snippets/system-text-json-how-to-5-0/csharp/PreserveReferences.cs" highlight="34":::
+
+Here are some other notes about preserving references:
+
+* `MaxDepth` validation is not affected.
+* The following types can be serialized as references but can't be deserialized as references:
+  * Immutable types
+  * Arrays
+  * Value types, such as structs
```

Comment:
> We should also mention that we throw when reference metadata is found when deserializing these types.

We throw if an Immutable type or Array contains $`ref` or `$id`.
POCO value types can ignore `$id` but will throw if they contain `$ref`.

Reply:
@tdykstra I think it would say: "this feature cannot be used to preserve value types and/or immutable types"
Reasons: 
* Value types: we don't have a reference per se.
* Immutable types: they are created until the payload is fully read, which makes it impossible to preserve a reference of itself that could be within the payload e.g: 
```cs
var options = new JsonSerializerOptions { ReferenceHandler = ReferenceHandler.Preserve };
var obj = JsonSerializer.Deserialize<ImmutableDictionary<string, object>>(@"{""$id"": ""1"", ""property1"": {""$ref"": ""1""} }", options);
```